src/gather.jl

```diff
@@ -0,0 +1,49 @@
+export gather, gather_indices
+
+"""
+    gather!(dst, src, idx)
+
+Reverse operation of scatter, which gather data in `src` to destination according to `idx`.
+For each index `k` in `idx`, assign values to `dst` according to
+
+    dst[k...] = src[idx[k]...]
+
+# Arguments
+- `dst`: the destination where data would be assigned to.
+- `src`: the source data to be assigned.
+- `idx`: the mapping for assignment from source to destination.
+"""
+function gather!(dst::AbstractArray{T,N}, src::AbstractArray{T}, idx::AbstractArray{<:Integer,N}) where {T,N}
+    @assert size(dst) == size(idx) "dst and idx must have the same size."
+    @simd for k = CartesianIndices(idx)
+        @inbounds view(dst, k) .= view(src, idx[k]...)
```

Comment:
`k` is only `Integer` here, so no `...` needed.

```suggestion
        @inbounds dst[k] = src[idx[k]]
```

Reply:
Shouldn't gather accept `idx::AbstractArray{<:IntOrTuple}` for symmetry? In which case the splat is needed. 

But the views seem strange, see my comment somewhere... unless the intention is to allow e.g. `idx = [(1,:), (3,:)]`?

Reply:
You are right, this should accept `IntOrTuple`, then the splat is actually needed. I'll update the suggestion.

I don't think something like `idx = [(1, :), (3,:)]` would be allowed. The semantic is unclear.

Reply:
`CartesianIndex` is not tuple, so it doesn't need unpacking `k...`.

Reply:
Check new commit please.